src/app/component/Swipeable/Swipeable.component.js

```diff
@@ -0,0 +1,126 @@
+import './Swipeable.style';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const THRESHOLD_PERC = 50;
+const TRANSITION_CSS = 'transition: transform 0.15s linear;';
+const PERCENT = 100;
+
+export class Swipeable extends React.Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        onRightSwipe: PropTypes.func.isRequired,
+        rightSwipeText: PropTypes.string.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.swipeObjRef = React.createRef();
+        this.leftChild = React.createRef();
+        this.rightChild = React.createRef();
+
+        this.startX = null;
+        this.deltaX = null;
+        this.storedOffset = 0;
+        this.elemWidth = null;
+
+        this.onTouchStart = this.onTouchStart.bind(this);
+        this.onTouchMove = this.onTouchMove.bind(this);
+        this.onTouchEnd = this.onTouchEnd.bind(this);
+    }
+
+    onTouchStart(e) {
+        this.startX = e.touches[0].pageX;
+
+        if (this.storedOffset !== 0) {
+            this.startX -= this.storedOffset;
+        }
+        this.storedOffset = 0;
+
+        this.elemWidth = this.swipeObjRef.current.getBoundingClientRect().width;
+
+        this.swipeObjRef.current.style = `transition: none 0; transform: ${this.swipeObjRef.current.style.transform}`;
+    }
+
+    getDeltaXPercent() {
+        return (this.deltaX / this.elemWidth) * PERCENT;
+    }
+
+    onTouchMove(e) {
+        this.deltaX = e.touches[0].pageX - this.startX;
+        if (this.deltaX > 0) {
+            this.deltaX = 0;
+        }
+
+        this.swipeObjRef.current.style.transform = `translateX( ${ this.getDeltaXPercent() }% )`;
```

Comment:
Consider implementing translate using CSS variables. Pass a percentage into it. See slider on Product page as an example.